api/app/controllers/webhook.controller.ts

```diff
@@ -318,17 +341,19 @@ export async function handlePushEvent(req: Request, res: Response) {
       .status(400)
       .json({ message: "Invalid branch for the current environment" });
   }
-
   try {
-    for (const commit of commits) {
-      await processCommit(commit, pushEvent, "added", branch);
-      await processCommit(commit, pushEvent, "modified", branch);
-    }
+    const commitPromises = commits.map(async (commit: any) => {
+      try {
+        await processCommit(commit, pushEvent, "added", branch);
+        await processCommit(commit, pushEvent, "modified", branch);
+      } catch (error: any) {
+        sendAlert(error.message, true);
+        return { status: "rejected", reason: error.message };
+      }
+    });
+    await Promise.allSettled(commitPromises);
   } catch (error) {
-    // Send error email
-    const message = error instanceof Error ? error.message : "Unknown error";
-    await sendAlert(message);
-    return res.status(500).json({ message: message });
+    return handleError(error, res);
   }
-  return res.status(200).json({ message: "Transcript queued Successfully" });
+  return res.sendStatus(200);
```

Comment:
any reason why we changed this?

Reply:
This is what is sent back to the GitHub webhook portal. I don't see it, if it runs well, 200 OK is enough response(IMO), if there is an error the error and its message are sent.

Reply:
ok